provisioning/service/src/HttpTransportSettings.cs

```diff
@@ -23,5 +23,11 @@ public HttpTransportSettings()
         /// Gets or sets proxy information for the request.
         /// </summary>
         public IWebProxy Proxy { get; set; }
+
+        /// <summary>
+        /// To enable certificate revocation check. Default to be false.
+        /// </summary>
+        public bool CertificateRevocationCheck { get; set; } = true;
```

Comment:
Wondering if this should be something you can send to the constructor and can be optional (set to true by default). When someone creates an instance, they will easily be able to tell what options are possible rather than going and check what can be set in the class. As a user, I would do new HttpTransportSettings() and it would be much easier for me if the intellisense shows that i can send this parameter.

Reply:
The question is, which of the many, many properties should be in the ctor? It gets too long and then the ctor is hard to use.

That is why, with option classes like this, the guidance is to rely on public properties that someone sets, ideally via initializer list.

Reply:
There were only two here but the explanation makes sense.